src/gluonts/model/deepvar_hierarchical/_network.py

```diff
@@ -84,63 +83,32 @@ def reconcile_samples(self, samples: Tensor) -> Tensor:
         ----------
         samples
             Unconstrained samples
+            Shape: (num_samples, batch_size, seq_len, target_dim) during training and
+                   (num_parallel_samples x batch_size, seq_len, target_dim) during prediction.
 
         Returns
         -------
         Coherent samples
             Tensor, shape same as that of `samples`.
 
         """
-        proj_matrix_shape = self.M.shape  # (num_ts, num_ts)
-
-        num_iter_dims = len(self.seq_axis) if self.seq_axis else 0
-
-        # Expand `M` depending on the shape of samples:
-        # If seq_axis = None, during training the first axis is only `batch_size`, in which case `M` would be expanded
-        # 3 times; during prediction it would be expanded 2 times since the first axis is
-        # `batch_size x num_parallel_samples`.
-        M_expanded = self.M
-        for i in range(len(samples.shape[num_iter_dims:-1])):
-            M_expanded = M_expanded.expand_dims(axis=0)
-
-        # If seq_axis = None broadcast M to (num_samples, batch_size, seq_len, m, m) during training
-        # and to (num_samples * batch_size, seq_len, m, m) during prediction
-        # Else broadcast to the appropriate remaining dimension
-        _shape = (
-            list(samples.shape[:-1])
-            if not self.seq_axis
-            else [
-                samples.shape[i]
-                for i in range(len(samples.shape[:-1]))
-                if i not in self.seq_axis
-            ]
-        )
-        self.M_broadcast = mx.nd.broadcast_to(
-            M_expanded,
-            shape=_shape + list(proj_matrix_shape),
-        )
-
         if self.seq_axis:
             # bring the axis to iterate in the beginning
             samples = mx.nd.moveaxis(
                 samples, self.seq_axis, list(range(len(self.seq_axis)))
             )
 
-            out = []
-            for idx in product(
-                *[
-                    range(x)
-                    for x in [
-                        samples.shape[d] for d in range(len(self.seq_axis))
+            out = [
```

Comment:
Not sure what this means and it seems you removed the comments explaining this in the PR? Maybe it is worth dropping the coment again/explaining this line.